src/net.cpp

```diff
@@ -1080,7 +1080,12 @@ void ThreadSocketHandler()
                 }
             }
         }
-        if(vNodes.size() != nPrevNodeCount) {
+        size_t vNodesSize;
+        {
+            LOCK(cs_vNodes);
+            vNodesSize = vNodes.size();
+        }
+        if(vNodesSize != nPrevNodeCount) {
             nPrevNodeCount = vNodes.size();
```

Comment:
Will have to close this as I am reforking the Dash repo and no longer have access to the patch-7 branch.